### PR TITLE
[release/v2.20] Use `registry.k8s.io` instead of `k8s.gcr.io` for Kubernetes upstream images

### PR DESCRIPTION
**Pull request by embik**
**What this PR does / why we need it**:
Kubernetes upstream introduced a new registry URL called registry.k8s.io, which acts as a kind of proxy (well, technically it only redirects) for upstream images. This can speed up the image pull process if a local mirror is available that you're forwarded to (mainly on AWS, as far as I understand from upstream documentation) and is the suggested source of images now. Let's switch to it.

This is a manual cherry-pick of #11079.

**Which issue(s) this PR fixes**:

Towards #9987

**What type of PR is this?**
/kind cleanup
/kind chore

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
ACTION REQUIRED: Use `registry.k8s.io` instead of `k8s.gcr.io` for Kubernetes upstream images. It might be necessary to update firewall rules or mirror registries accordingly
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

**Review by xrstf — APPROVED** [review]
/approve

**Comment by embik** [comment]
/hold

needs a final decision whether to be included or not.

**Review by ahmedwaleedmalik — COMMENTED** [review]
/approve 👍

**Pull request merged**
